src/components/forms/components/SearchSelectionMenu.js

```diff
@@ -1,52 +1,110 @@
 import React, { useEffect, useRef } from 'react';
 import PropTypes from 'prop-types';
 import { Dropdown } from 'react-bootstrap';
+import { VerticalScrollContainer } from './VerticalScrollContainer';
 
 
 export class SearchSelectionMenu extends React.PureComponent {
+
+    static defaultProps = {
+        titleRenderFunction: function(option){
+            return option;
+        }
+    };
+
     constructor(props){
         super(props);
-        this.onToggleOpen = this.onToggleOpen.bind(this);
         this.state = {
-            dropOpen: false
+            dropOpen: false,
+            refreshKey: 0 // iterated to force a refresh of dropdown
```

Comment:
`// incremented ..` :-P

Is this (+ set in componentDidUpdate) ~~really necessary (since it seemed to auto-re-layout on scroll) or this handle other things?~~ bleh sorry I need to read before typing, realizing is re: options changing.

Not important but for fun might consider ways to make this slightly more performant if possible, i.e. if we have a max height that can only fit ~ 20 options, could do like if `Math.min(20, oldOpts.length) !== Math.min(20, newOpts.length)` to reduce # of state updates so that # of results change from like 97 to 340, and it doesnt visually impact dropdown size, can skip the increment. Idk probably not the added code since likely is already more than performant enough.

Another thought (though unsure if better) is can maybe call `this.forceUpdate()` to trigger a refresh of component without needing an additional state field (refreshKey).

3rd thought (probably wouldn't go this route mostly because we don't have control of React-Bootstrap and their API or whatever might change) is maybe if could pass down refs to React-Bootstrap's elements, access popper instance, and then call `popper.relayout()` or something or other. Since PopperJS isn't a React thing, theoretically for perf. gain should be fine to bypass React (aside from detecting componentDidUpdate) to manage it.

Reply:
Whoops on the incremented. 

The problem I have with max heights is that once we add in more "customized item selections" with data for the specific object (showing age, gender for individuals, for instance), that max height per dropdown item won't always be consistent, so we'll have to access one of the nodes directly. I'm not 100% sure, but I'm thinking that could end up being really complicated due to the way that `SearchSelectionMenu` is implemented. Since `Dropdown.Item`s are all rendered inside of a map function, even if we were able to assign refs to one/some of them, they'd end up being references to React components, and I'm not sure how to get the height of the rendered html element out of that... would we use findDomNode or something?

I was thinking about using forceUpdate, but for the reasons listed [here](https://stackoverflow.com/questions/30626030/can-you-force-a-react-component-to-rerender-without-calling-setstate) didn't do so. Also found [an issue](https://github.com/mui-org/material-ui/issues/13179) with almost the exact same problem, and they ended up going this way, as well. So it seems like this is at least a somewhat standard way of handling this.

Yeah, I considered that too when I saw that in Popper's docs, but I'm thinking that anything that requires going through React-Bootstrap to get to the Popper element is probably a no-go. React-Bootstrap doesn't seem to give us a clean-cut way to access the Popper instance through their API; this is something I ran into when I thought the issue was a Popper ref issue and was trying to find a way to check exactly what reference & popper elements were being passed into the Popper element.

So yeah, IDK. I'm not sure what the best way to limit the re-renders would be. In the meantime we can probably come back to it if it becomes an issue.